cmd/controller/app/app.go

```diff
@@ -296,48 +289,36 @@ func (a *App) runKubeServer(ctx context.Context, log *logging.Logger, client *ku
 }
 
 func (a *App) runMetricsHTTPServer(ctx context.Context, log *logging.Logger) error {
-	e := echo.New()
-	e.HideBanner = true
-	e.Debug = false
+	log.Info("running http server")
+	defer log.Info("stopping http server")
 
-	e.Use(middleware.Recover())
-	e.GET("/metrics", echo.WrapHandler(promhttp.Handler()))
-	e.GET("/debug/pprof/*item", echo.WrapHandler(http.DefaultServeMux))
+	mux := http.NewServeMux()
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte("ok"))
+	})
+	mux.Handle("/metrics", promhttp.Handler())
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 	srv := http.Server{
 		Addr:         fmt.Sprintf(":%d", a.cfg.MetricsHTTPListenPort),
-		Handler:      e,
-		ReadTimeout:  10 * time.Second,
+		Handler:      mux,
+		ReadTimeout:  1 * time.Minute,
 		WriteTimeout: 1 * time.Minute,
 	}
+
 	go func() {
 		<-ctx.Done()
-		log.Info("shutting metrics down http server")
 		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 		defer cancel()
-		if err := srv.Shutdown(ctx); err != nil {
-			log.Error(err.Error())
-		}
+		_ = srv.Shutdown(ctx)
 	}()
-	log.Infof("running metrics server, port=%d", a.cfg.MetricsHTTPListenPort)
+
 	if err := srv.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
-		return err
+		return fmt.Errorf("http serve: %w", err)
 	}
-	return nil
-}
 
-func withPyroscope(addr string) {
```

Comment:
Why was pyroscope removed?

Reply:
it's not longer needed as library which adds extra binary size. Now it works well with annotation.s